ddp-workspace/projects/ddp-dsm-ui/src/app/activity-data/activity-data.component.html

```diff
@@ -1,131 +1,141 @@
+    <div *ngIf="answer">
+      <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+      <p><b>{{ answer }}</b></p>
+    </div>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'NUMERIC'">
+    <div *ngIf="answer">
+      <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+      <p><b>{{ answer }}</b></p>
+    </div>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'PICKLIST'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+
+    <div *ngIf="picklist.multiple">
+      <ul>
+        <li *ngFor="let option of picklist.multiple.options">
+
+          <span *ngIf="option.text"><b>Text:</b> {{ option.text }}</span>
+          <br/>
+
+          <ul>
+            <li *ngFor="let nt of option.nText"><b>nText:</b> {{ nt }}</li>
+          </ul>
+          <span *ngIf="option.details"><b>Details:</b> {{ option.details }}</span>
+          <br/>
+        </li>
+
+      </ul>
+
+      <ul>
+        <li *ngFor="let group of picklist.multiple.groups">
+
+          <span *ngIf="group.text"><b>Text (Group):</b> {{ group.text }}</span>
+          <br/>
+
+          <ul>
+            <li *ngFor="let optText of group.optionText"><b>Option text (Group):</b> {{ optText }}</li>
+          </ul>
+          <br/>
+
+          <ul>
+            <li *ngFor="let gd of group.details"><b>Details (Group):</b> {{ gd }}</li>
+          </ul>
+
+        </li>
+      </ul>
+    </div>
+    <div *ngIf="picklist.single">
+      <ul>
+        <li *ngFor="let single of picklist.single">
+          <span *ngIf="single.text"><b>Text:</b> {{ single.text }}</span>
+          <br/>
+          <span *ngIf="single.details"><b>Details:</b> {{ single.details }}</span>
+        </li>
+      </ul>
+    </div>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'COMPOSITE'">
+    <h5 style="color: #A9A9A9"  innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <ul>
+      <li *ngFor="let compose of composite"><b *ngIf="compose.question">{{compose.question}}:</b>
+        <span *ngIf="compose.answer">{{compose.answer}}</span></li>
+    </ul>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'BOOLEAN'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <p><b>{{ answer }}</b></p>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'AGREEMENT'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <p><b>{{ answer }}</b></p>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'MATRIX'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <ul>
+      <li *ngFor="let mat of matrix"><b>{{mat.question}}:</b> {{mat.answer}}</li>
+    </ul>
+  </ng-container>
+
+  <ng-container *ngIf="type === 'DATE'">
+    <h5 style="color: #A9A9A9" innerHTML="{{question}} <b>({{stableId}})</b>"></h5>
+    <p><b>{{ answer }}</b></p>
+  </ng-container>
+
+</ng-template>
```

Comment:
The overall file looks good, I'm just wondering, shouldn't we get the "nice" text from the QuestionDefinition for all the texts that we are displaying?

Reply:
No that is fine, I saw that you are using the ActivityDefinition with the buildActivity method, right?